cmd/resim/commands/styling.go

```diff
@@ -59,15 +61,82 @@ var ReSimUsageTemplate string = `{{StyleHeading "USAGE"}}{{if .Runnable}}
 Use "{{.CommandPath}} [command] --help" for more information about a command.{{end}}
 `
 
-var templateFuncs = template.FuncMap{
+var FlagSetUsageTemplate string = `{{if .HasRequired}}
+
+{{StyleHeading "REQUIRED FLAGS"}}
+{{.RequiredUsages}}{{end}}{{if .HasOptional}}
+
+{{StyleHeading "OPTIONAL FLAGS"}}
+{{.OptionalUsages}}{{end}}
+`
+
+var flagTemplateFuncs = template.FuncMap{
 	"StyleHeading": styleHeading,
 }
 
+var resimTemplateFuncs = template.FuncMap{
+	"StyleHeading":  styleHeading,
+	"FlagSetUsages": flagSetUsages,
+}
+
+var BashCompOneRequiredFlag string = "cobra_annotation_bash_completion_one_required_flag"
+
+type FlagSets struct {
+	requiredFlags pflag.FlagSet
+	optionalFlags pflag.FlagSet
+}
+
+func (fs FlagSets) HasRequired() bool {
+	return fs.requiredFlags.HasFlags()
+}
+
+func (fs FlagSets) RequiredUsages() string {
+	return strings.TrimRightFunc(fs.requiredFlags.FlagUsages(), unicode.IsSpace)
+}
+
+func (fs FlagSets) HasOptional() bool {
+	return fs.optionalFlags.HasFlags()
+}
+
+func (fs FlagSets) OptionalUsages() string {
+	return strings.TrimRightFunc(fs.optionalFlags.FlagUsages(), unicode.IsSpace)
+}
+
+func flagSetsBuilder(flags *pflag.FlagSet) FlagSets {
+	var allFlags FlagSets
+	flags.VisitAll(func(flag *pflag.Flag) {
+		requiredAnnotation, found := flag.Annotations[BashCompOneRequiredFlag]
+		if !found {
+			allFlags.optionalFlags.AddFlag(flag)
+			return
```

Comment:
naked return - probably best to be explicit

Reply:
This function has a void return. So I am not sure what to do here. Again - go novice here. As I don't know the right way to do this I am going to leave a TODO ticket.